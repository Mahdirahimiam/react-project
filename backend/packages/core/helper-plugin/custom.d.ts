export {};

declare global {
  interface Window {
    strapi: {
      backendURL: string;
      telemetryDisabled: boolean;
      projectType: 'Enterprise' | 'Community';
    };
  }
}

import { StrapiTheme } from '@strapi/design-system';

declare module 'styled-components' {
  // eslint-disable-next-line @typescript-eslint/no-empty-interface
  export interface DefaultTheme extends StrapiTheme {}
}
