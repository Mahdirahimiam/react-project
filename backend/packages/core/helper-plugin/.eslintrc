{
  "root": true,
  "extends": ["custom/front/typescript", "plugin:storybook/recommended"],
  "rules": {
    "import/no-unresolved": [
      "error",
      {
        // Eslint does not detect the package.json exports field
        // and reports broken imports for the following imports
        "ignore": ["@strapi/design-system/v2"]
      }
    ]
  },
  "overrides": [
    {
      "files": ["./jest.config.front.js", "./webpack.config.js"],
      "env": {
        "node": true
      },
      "rules": {
        "@typescript-eslint/no-var-requires": "off"
      }
    },
    {
      "files": ["./tests/*"],
      "env": {
        "jest": true
      }
    }
  ]
}
