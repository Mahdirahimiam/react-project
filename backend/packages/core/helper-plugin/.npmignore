############################
# dts
############################

dist/**/*.d.ts.map