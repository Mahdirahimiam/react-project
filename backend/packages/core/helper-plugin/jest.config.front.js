module.exports = {
  preset: '../../../jest-preset.front.js',
  collectCoverageFrom: ['<rootDir>/packages/core/helper-plugin/src/**/*.js'],
  displayName: 'Helper plugin',
  moduleNameMapper: {
    '^@tests/(.*)$': '<rootDir>/tests/$1',
  },
  setupFilesAfterEnv: ['./tests/setup.ts'],
};
