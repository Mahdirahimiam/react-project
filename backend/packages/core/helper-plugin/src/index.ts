/* -------------------------------------------------------------------------------------------------
 * Components
 * -----------------------------------------------------------------------------------------------*/

export * from './components/AnErrorOccurred';
export * from './components/CheckPagePermissions';
export * from './components/CheckPermissions';
export * from './components/ConfirmDialog';
export * from './components/ContentBox';
export * from './components/DateTimePicker';
export * from './components/DynamicTable';
export * from './components/Table';
export * from './components/EmptyStateLayout';
export * from './components/EmptyBodyTable';
export * from './components/FilterListURLQuery';
export * from './components/FilterPopoverURLQuery';
export * from './components/Form';
export * from './components/GenericInput';
export * from './components/InjectionZone';
export * from './components/Link';
export * from './components/LinkButton';
export * from './components/LoadingIndicatorPage';
export * from './components/NoContent';
export * from './components/NoMedia';
export * from './components/NoPermissions';
export * from './components/NotAllowedInput';
export * from './components/PageSizeURLQuery';
export * from './components/PaginationURLQuery';
export * from './components/ReactSelect';
export * from './components/RelativeTime';
export * from './components/SearchURLQuery';
export * from './components/SettingsPageTitle';
export * from './components/Status';

/* -------------------------------------------------------------------------------------------------
 * Content Manager
 * -----------------------------------------------------------------------------------------------*/

export {
  useCMEditViewDataManager,
  ContentManagerEditViewDataManagerContext,
  type CMEditViewDataManagerContextValue,
  type ContentType,
} from './content-manager/CMEditViewDataManager';
export * from './content-manager/utils/getAttributeInfos';

/* -------------------------------------------------------------------------------------------------
 * Features
 * -----------------------------------------------------------------------------------------------*/

export * from './features/AppInfo';
export * from './features/AutoReloadOverlayBlocker';
export * from './features/CustomFields';
export * from './features/GuidedTour';
export * from './features/Library';
export * from './features/Notifications';
export * from './features/OverlayBlocker';
export * from './features/RBAC';
export * from './features/StrapiApp';
export * from './features/Tracking';

/* -------------------------------------------------------------------------------------------------
 * Hooks
 * -----------------------------------------------------------------------------------------------*/

export * from './hooks/useAPIErrorHandler';
export * from './hooks/useCallbackRef';
export * from './hooks/useClipboard';
export * from './hooks/useCollator';
export * from './hooks/useFetchClient';
export * from './hooks/useFilter';
export * from './hooks/useFocusInputField';
export * from './hooks/useFocusWhenNavigate';
export * from './hooks/useLockScroll';
export * from './hooks/usePersistentState';
export * from './hooks/useQuery';
export * from './hooks/useQueryParams';
export * from './hooks/useRBAC';
export * from './hooks/useSelectionState';

/* -------------------------------------------------------------------------------------------------
 * Icons
 * -----------------------------------------------------------------------------------------------*/

export * from './icons/RemoveRoundedButton';
export * from './icons/SortIcon';

/* -------------------------------------------------------------------------------------------------
 * Utils
 * -----------------------------------------------------------------------------------------------*/

export * from './content-manager/utils/contentManagementUtilRemoveFieldsFromData';
export * from './content-manager/utils/formatContentTypeData';
export * from './utils/auth';
export * from './utils/awaitToJs';
export * from './utils/difference';
export * from './utils/getAPIInnerErrors';
export * from './utils/getFetchClient';
export * from './utils/getFileExtension';
export * from './utils/getYupInnerErrors';
export * from './utils/hasPermissions';
export * from './utils/normalizeAPIError';
export * from './utils/prefixFileUrlWithBackendUrl';
export * from './utils/prefixPluginTranslations';
export * from './utils/pxToRem';
export * from './utils/request';
export * from './utils/setHexOpacity';
export * from './utils/stopPropagation';
export * from './utils/translatedErrors';
export * from './utils/wrapAxiosInstance';

/* -------------------------------------------------------------------------------------------------
 * Types
 * -----------------------------------------------------------------------------------------------*/

export type { TranslationMessage, FilterData } from './types';
