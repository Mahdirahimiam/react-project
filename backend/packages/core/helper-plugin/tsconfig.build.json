{
  "extends": "./tsconfig.json",
  "exclude": ["**/*.test.tsx", "**/*.test.ts"],
  "compilerOptions": {
    "declarationMap": true,
    "outDir": "./dist"
  }
}
