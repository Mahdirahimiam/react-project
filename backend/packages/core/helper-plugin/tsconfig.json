{
  "extends": "tsconfig/client.json",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@tests/*": ["./tests/*"]
    }
  },
  "include": ["src", "custom.d.ts"]
}
