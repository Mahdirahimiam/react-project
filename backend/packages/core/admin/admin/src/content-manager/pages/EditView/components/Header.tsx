import * as React from 'react';

import {
  Box,
  Button,
  Dialog,
  DialogBody,
  DialogFooter,
  Flex,
  HeaderLayout,
  Typography,
} from '@strapi/design-system';
import { Link } from '@strapi/design-system/v2';
import { AllowedActions, useCMEditViewDataManager } from '@strapi/helper-plugin';
import { ArrowLeft, Check, ExclamationMarkCircle } from '@strapi/icons';
import get from 'lodash/get';
import isEqual from 'lodash/isEqual';
import { useIntl } from 'react-intl';
import { NavLink, useHistory } from 'react-router-dom';

import { getTranslation } from '../../../utils/translations';

interface HeaderProps {
  allowedActions: AllowedActions;
}

const Header = ({ allowedActions: { canUpdate, canCreate, canPublish } }: HeaderProps) => {
  const {
    initialData,
    isCreatingEntry,
    isSingleType,
    status,
    layout,
    hasDraftAndPublish,
    modifiedData,
    onPublish,
    onUnpublish,
    publishConfirmation: { show: showPublishConfirmation, draftCount } = {},
    onPublishPromptDismissal,
  } = useCMEditViewDataManager();
  const { goBack } = useHistory();
  const [showWarningUnpublish, setWarningUnpublish] = React.useState(false);
  const { formatMessage } = useIntl();

  const currentContentTypeMainField = get(layout, ['settings', 'mainField'], 'id');
  const currentContentTypeName = get(layout, ['info', 'displayName'], 'NOT FOUND');
  const didChangeData =
    !isEqual(initialData, modifiedData) ||
    (isCreatingEntry && Object.keys(modifiedData).length > 0);

  const createEntryIntlTitle = formatMessage({
    id: getTranslation('containers.Edit.pluginHeader.title.new'),
    defaultMessage: 'Create an entry',
  });

  let title = createEntryIntlTitle;

  if (!isCreatingEntry && !isSingleType) {
    title = initialData[currentContentTypeMainField] || currentContentTypeName;
  }

  if (isSingleType) {
    title = currentContentTypeName;
  }

  let primaryAction = null;

  if (isCreatingEntry && canCreate) {
    primaryAction = (
      <Flex gap={2}>
        {hasDraftAndPublish && (
          <Button disabled startIcon={<Check />} variant="secondary">
            {formatMessage({ id: 'app.utils.publish', defaultMessage: 'Publish' })}
          </Button>
        )}
        <Button disabled={!didChangeData} loading={status === 'submit-pending'} type="submit">
          {formatMessage({
            id: getTranslation('containers.Edit.submit'),
            defaultMessage: 'Save',
          })}
        </Button>
      </Flex>
    );
  }

  if (!isCreatingEntry && canUpdate) {
    const shouldShowPublishButton = hasDraftAndPublish && canPublish;
    const isPublished = typeof initialData.publishedAt === 'string';
    const isPublishButtonLoading = isPublished
      ? status === 'unpublish-pending'
      : status === 'publish-pending';
    const pubishButtonLabel = isPublished
      ? { id: 'app.utils.unpublish', defaultMessage: 'Unpublish' }
      : { id: 'app.utils.publish', defaultMessage: 'Publish' };

    const onClick = isPublished ? () => setWarningUnpublish(true) : () => onPublish?.();

    primaryAction = (
      <Flex>
        {shouldShowPublishButton && (
          <Button
            disabled={didChangeData}
            loading={isPublishButtonLoading}
            onClick={onClick}
            startIcon={<Check />}
            variant="secondary"
          >
            {formatMessage(pubishButtonLabel)}
          </Button>
        )}
        <Box paddingLeft={shouldShowPublishButton ? 2 : 0}>
          <Button disabled={!didChangeData} loading={status === 'submit-pending'} type="submit">
            {formatMessage({
              id: getTranslation('containers.Edit.submit'),
              defaultMessage: 'Save',
            })}
          </Button>
        </Box>
      </Flex>
    );
  }

  const toggleWarningUnpublish = () => setWarningUnpublish((prevState) => !prevState);

  const handleUnpublish = () => {
    toggleWarningUnpublish();
    onUnpublish?.();
  };

  const subtitle = `${formatMessage({
    id: getTranslation('api.id'),
    defaultMessage: 'API ID',
    // @ts-expect-error – issue comes from the context not having the correct layout from the admin.
  })}: ${layout?.apiID}`;

  return (
    <>
      <HeaderLayout
        title={title.toString()}
        primaryAction={primaryAction}
        subtitle={subtitle}
        navigationAction={
          <Link
            startIcon={<ArrowLeft />}
            // Needed in order to redirect the user with the correct search params
            // Since parts is using a link from react-router-dom the best way to do it is to disable the
            // event
            onClick={(e) => {
              e.preventDefault();
              goBack();
            }}
            as={NavLink}
            // @ts-expect-error – DS issue with inferring props from the `as` prop.
            to="/"
          >
            {formatMessage({
              id: 'global.back',
              defaultMessage: 'Back',
            })}
          </Link>
        }
      />
      <Dialog
        onClose={toggleWarningUnpublish}
        title="Confirmation"
        aria-labelledby="confirmation"
        aria-describedby="confirm-description"
        isOpen={showWarningUnpublish}
      >
        <DialogBody icon={<ExclamationMarkCircle />}>
          <Flex direction="column" alignItems="stretch" gap={2}>
            <Flex justifyContent="center" style={{ textAlign: 'center' }}>
              <Typography id="confirm-description">
                {formatMessage({
                  id: getTranslation('popUpWarning.warning.unpublish'),
                  defaultMessage: 'Unpublish this content will automatically change it to a draft.',
                })}
              </Typography>
            </Flex>
            <Flex justifyContent="center" style={{ textAlign: 'center' }}>
              <Typography id="confirm-description">
                {formatMessage({
                  id: getTranslation('popUpWarning.warning.unpublish-question'),
                  defaultMessage: 'Are you sure you want to unpublish it?',
                })}
              </Typography>
            </Flex>
          </Flex>
        </DialogBody>
        <DialogFooter
          startAction={
            <Button onClick={toggleWarningUnpublish} variant="tertiary">
              {formatMessage({
                id: 'components.popUpWarning.button.cancel',
                defaultMessage: 'Cancel',
              })}
            </Button>
          }
          endAction={
            <Button variant="danger-light" onClick={handleUnpublish}>
              {formatMessage({
                id: 'components.popUpWarning.button.confirm',
                defaultMessage: 'Confirm',
              })}
            </Button>
          }
        />
      </Dialog>
      <Dialog
        // @ts-expect-error – Context issue with it living in the helper-plugin
        onClose={onPublishPromptDismissal}
        title={formatMessage({
          id: getTranslation(`popUpWarning.warning.has-draft-relations.title`),
          defaultMessage: 'Confirmation',
        })}
        labelledBy="confirmation"
        describedBy="confirm-description"
        // @ts-expect-error – Context issue with it living in the helper-plugin
        isOpen={showPublishConfirmation}
      >
        <DialogBody icon={<ExclamationMarkCircle />}>
          <Flex direction="column" alignItems="stretch" gap={2}>
            <Typography textAlign="center" id="confirm-description">
              {draftCount}
              {/* @ts-expect-error – this is an issue with rendering a component with the formatMessage helper, perhaps a mis-match in types? */}
              {formatMessage(
                {
                  id: getTranslation(`popUpwarning.warning.has-draft-relations.message`),
                  defaultMessage:
                    '<b>{count, plural, one { relation is} other { relations are}}</b> not published yet and might lead to unexpected behavior.',
                },
                {
                  b: (chunks) => <Typography fontWeight="bold">{chunks}</Typography>,
                  count: draftCount,
                }
              )}
            </Typography>
            <Typography textAlign="center" id="confirm-description">
              {formatMessage({
                id: getTranslation('popUpWarning.warning.publish-question'),
                defaultMessage: 'Do you still want to publish?',
              })}
            </Typography>
          </Flex>
        </DialogBody>
        <DialogFooter
          startAction={
            <Button onClick={onPublishPromptDismissal} variant="tertiary">
              {formatMessage({
                id: 'components.popUpWarning.button.cancel',
                defaultMessage: 'Cancel',
              })}
            </Button>
          }
          endAction={
            <Button variant="success" onClick={onPublish}>
              {formatMessage({
                id: getTranslation('popUpwarning.warning.has-draft-relations.button-confirm'),
                defaultMessage: 'Publish',
              })}
            </Button>
          }
        />
      </Dialog>
    </>
  );
};

export { Header };
export type { HeaderProps };
