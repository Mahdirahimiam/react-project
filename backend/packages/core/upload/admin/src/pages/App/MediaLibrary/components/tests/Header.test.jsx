import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { useQueryParams } from '@strapi/helper-plugin';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MemoryRouter } from 'react-router-dom';

import { Header } from '../Header';

jest.mock('@strapi/helper-plugin', () => ({
  ...jest.requireActual('@strapi/helper-plugin'),
  useQueryParams: jest.fn().mockReturnValue([{ query: {}, rawQuery: '' }, jest.fn()]),
}));

const FIXTURE_FOLDER = {
  id: 1,
  name: 'Folder 1',
  children: {
    count: 1,
  },
  createdAt: '',
  files: {
    count: 1,
  },
  path: '/1',
  pathId: 1,
  updatedAt: '',
  parent: {
    id: 2,
    name: 'Folder 2',
    children: {
      count: 1,
    },
    createdAt: '',
    files: {
      count: 1,
    },
    path: '/1',
    pathId: 1,
    updatedAt: '',
  },
};

const setup = (props) => {
  const withDefaults = {
    ...props,
    assetCount: 2,
    folderCount: 2,
    canCreate: true,
    onToggleEditFolderDialog: jest.fn(),
    onToggleUploadAssetDialog: jest.fn(),
  };

  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        refetchOnWindowFocus: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={lightTheme}>
        <IntlProvider locale="en">
          <MemoryRouter>
            <Header {...withDefaults} />
          </MemoryRouter>
        </IntlProvider>
      </ThemeProvider>
    </QueryClientProvider>
  );
};

describe('Header', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders', () => {
    const { container } = setup({ folder: FIXTURE_FOLDER });

    expect(container).toMatchSnapshot();
  });

  test('does not render a back button at the root level of the media library', () => {
    const { queryByText } = setup({ folder: null });

    expect(queryByText('Back')).not.toBeInTheDocument();
  });

  test('does render a back button at a nested level of the media library', () => {
    useQueryParams.mockReturnValueOnce([{ rawQuery: '', query: { folder: 2 } }, jest.fn()]);
    const { queryByText } = setup({ folder: FIXTURE_FOLDER });

    expect(queryByText('Back')).toBeInTheDocument();
  });
});
