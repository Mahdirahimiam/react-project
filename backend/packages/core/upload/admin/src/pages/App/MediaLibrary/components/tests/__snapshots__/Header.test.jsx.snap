// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
.c17 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c6 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #32324d;
}

.c10 {
  font-weight: 600;
  font-size: 2rem;
  line-height: 1.25;
  color: #32324d;
}

.c15 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #ffffff;
}

.c0 {
  background: #f6f6f9;
  padding-top: 24px;
  padding-right: 56px;
  padding-bottom: 40px;
  padding-left: 56px;
}

.c1 {
  padding-bottom: 8px;
}

.c3 {
  padding-right: 8px;
}

.c8 {
  min-width: 0;
}

.c12 {
  background: #4945ff;
  padding: 8px;
  padding-right: 16px;
  padding-left: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c13 {
  position: relative;
  outline: none;
}

.c13 > svg {
  height: 12px;
  width: 12px;
}

.c13 > svg > g,
.c13 > svg path {
  fill: #ffffff;
}

.c13[aria-disabled='true'] {
  pointer-events: none;
}

.c13:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c13:focus-visible {
  outline: none;
}

.c13:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c14 {
  height: 2rem;
  border: 1px solid #d9d8ff;
  background: #f0f0ff;
}

.c14 svg {
  height: 0.75rem;
  width: auto;
}

.c14[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c14[aria-disabled='true'] .c5 {
  color: #666687;
}

.c14[aria-disabled='true'] svg > g,
.c14[aria-disabled='true'] svg path {
  fill: #666687;
}

.c14[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c14[aria-disabled='true']:active .c5 {
  color: #666687;
}

.c14[aria-disabled='true']:active svg > g,
.c14[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c14:hover {
  background-color: #ffffff;
}

.c14:active {
  background-color: #ffffff;
  border: 1px solid #4945ff;
}

.c14:active .c5 {
  color: #4945ff;
}

.c14:active svg > g,
.c14:active svg path {
  fill: #4945ff;
}

.c14 .c5 {
  color: #271fe0;
}

.c14 svg > g,
.c14 svg path {
  fill: #271fe0;
}

.c16 {
  height: 2rem;
}

.c16 svg {
  height: 0.75rem;
  width: auto;
}

.c16[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c16[aria-disabled='true'] .c5 {
  color: #666687;
}

.c16[aria-disabled='true'] svg > g,
.c16[aria-disabled='true'] svg path {
  fill: #666687;
}

.c16[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c16[aria-disabled='true']:active .c5 {
  color: #666687;
}

.c16[aria-disabled='true']:active svg > g,
.c16[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c16:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c16:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c16 svg > g,
.c16 svg path {
  fill: #ffffff;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #4945ff;
  position: relative;
  outline: none;
}

.c2 svg path {
  -webkit-transition: fill 150ms ease-out;
  transition: fill 150ms ease-out;
  fill: currentColor;
}

.c2 svg {
  font-size: 0.625rem;
}

.c2 .c5 {
  -webkit-transition: color 150ms ease-out;
  transition: color 150ms ease-out;
  color: currentColor;
}

.c2:hover {
  color: #7b79ff;
}

.c2:active {
  color: #271fe0;
}

.c2:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c2:focus-visible {
  outline: none;
}

.c2:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    style="height: 0px;"
  >
    <div
      class="c0"
      data-strapi-header="true"
    >
      <div
        class="c1"
      >
        <a
          aria-current="page"
          class="c2 active"
          href="/?folder=2&folderPath=/1"
        >
          <span
            aria-hidden="true"
            class="c3 c4"
          >
            <svg
              fill="none"
              height="1rem"
              viewBox="0 0 24 24"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 13.3a.2.2 0 0 1-.2.2H5.74l8.239 8.239a.2.2 0 0 1 0 .282L12.14 23.86a.2.2 0 0 1-.282 0L.14 12.14a.2.2 0 0 1 0-.282L11.86.14a.2.2 0 0 1 .282 0L13.98 1.98a.2.2 0 0 1 0 .282L5.74 10.5H23.8c.11 0 .2.09.2.2v2.6Z"
                fill="#212134"
              />
            </svg>
          </span>
          <span
            class="c5 c6"
          >
            Back
          </span>
        </a>
      </div>
      <div
        class="c7"
      >
        <div
          class="c8 c9"
        >
          <h1
            class="c5 c10"
          >
            Media Library
          </h1>
        </div>
        <div
          class="c11"
        >
          <button
            aria-disabled="false"
            class="c12 c11 c13 c14"
            type="button"
          >
            <div
              aria-hidden="true"
              class=""
            >
              <svg
                fill="none"
                height="1rem"
                viewBox="0 0 24 24"
                width="1rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24 13.604a.3.3 0 0 1-.3.3h-9.795V23.7a.3.3 0 0 1-.3.3h-3.21a.3.3 0 0 1-.3-.3v-9.795H.3a.3.3 0 0 1-.3-.3v-3.21a.3.3 0 0 1 .3-.3h9.795V.3a.3.3 0 0 1 .3-.3h3.21a.3.3 0 0 1 .3.3v9.795H23.7a.3.3 0 0 1 .3.3v3.21Z"
                  fill="#212134"
                />
              </svg>
            </div>
            <span
              class="c5 c15"
            >
              Add new folder
            </span>
          </button>
          <button
            aria-disabled="false"
            class="c12 c11 c13 c16"
            type="button"
          >
            <div
              aria-hidden="true"
              class=""
            >
              <svg
                fill="none"
                height="1rem"
                viewBox="0 0 24 24"
                width="1rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24 13.604a.3.3 0 0 1-.3.3h-9.795V23.7a.3.3 0 0 1-.3.3h-3.21a.3.3 0 0 1-.3-.3v-9.795H.3a.3.3 0 0 1-.3-.3v-3.21a.3.3 0 0 1 .3-.3h9.795V.3a.3.3 0 0 1 .3-.3h3.21a.3.3 0 0 1 .3.3v9.795H23.7a.3.3 0 0 1 .3.3v3.21Z"
                  fill="#212134"
                />
              </svg>
            </div>
            <span
              class="c5 c15"
            >
              Add new assets
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c17"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
